## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [2]:
# Write your code for importing the csv file here
movies = pd.read_csv('./Movies.csv')
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [3]:
# Write your code for inspection here
movies.shape

(3853, 28)

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [4]:
movies.isna().sum()

color                         2
director_name                 0
num_critic_for_reviews        1
duration                      1
director_facebook_likes       0
actor_3_facebook_likes        6
actor_2_name                  1
actor_1_facebook_likes        0
gross                         0
genres                        0
actor_1_name                  0
movie_title                   0
num_voted_users               0
cast_total_facebook_likes     0
actor_3_name                  6
facenumber_in_poster          6
plot_keywords                30
movie_imdb_link               0
num_user_for_reviews          0
language                      4
country                       0
content_rating               48
budget                        0
title_year                    0
actor_2_facebook_likes        1
imdb_score                    0
aspect_ratio                 72
movie_facebook_likes          0
dtype: int64

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [5]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations
movies.drop(
    ["color", "director_facebook_likes", "actor_1_facebook_likes", "actor_2_facebook_likes", "actor_3_facebook_likes", "actor_2_name", "cast_total_facebook_likes", "actor_3_name", "duration", "facenumber_in_poster", "content_rating", "country", "movie_imdb_link", "aspect_ratio", "plot_keywords"],
axis=1)


director_name  num_critic_for_reviews        gross  \
0         James Cameron                   723.0  760505847.0   
1        Gore Verbinski                   302.0  309404152.0   
2            Sam Mendes                   602.0  200074175.0   
3     Christopher Nolan                   813.0  448130642.0   
4        Andrew Stanton                   462.0   73058679.0   
...                 ...                     ...          ...   
3848      Shane Carruth                   143.0     424760.0   
3849   Neill Dela Llana                    35.0      70071.0   
3850   Robert Rodriguez                    56.0    2040920.0   
3851       Edward Burns                    14.0       4584.0   
3852           Jon Gunn                    43.0      85222.0   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language       budget  title_year  imdb_score  \
0                   3054.0  English  237000000.0      2009.0         7.9   
1                   1238.0  English  300000000.0      2007.0         7.1   
2                    994.0  English  245000000.0      2015.0         6.8   
3                   2701.0  English  250000000.0      2012.0         8.5   
4                    738.0  English  263700000.0      2012.0         6.6   
...                    ...      ...          ...         ...         ...   
3848                 371.0  English       7000.0      2004.0         7.0   
3849                  35.0  English       7000.0      2005.0         6.3   
3850                 130.0  Spanish       7000.0      1992.0         6.9   
3851                  14.0  English       9000.0      2011.0         6.4   
3852                  84.0  English       1100.0      2004.0         6.6   

      movie_facebook_likes  
0                    33000  
1                        0  
2                    85000  
3                   164000  
4                    24000  
...                    ...  
3848                 19000  
3849                    74  
3850                     0  
3851                   413  
3852                   456  

[3853 rows x 13 columns]

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [6]:
# Write you code here
movies.isnull().sum()

color                         2
director_name                 0
num_critic_for_reviews        1
duration                      1
director_facebook_likes       0
actor_3_facebook_likes        6
actor_2_name                  1
actor_1_facebook_likes        0
gross                         0
genres                        0
actor_1_name                  0
movie_title                   0
num_voted_users               0
cast_total_facebook_likes     0
actor_3_name                  6
facenumber_in_poster          6
plot_keywords                30
movie_imdb_link               0
num_user_for_reviews          0
language                      4
country                       0
content_rating               48
budget                        0
title_year                    0
actor_2_facebook_likes        1
imdb_score                    0
aspect_ratio                 72
movie_facebook_likes          0
dtype: int64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [7]:
# Write your code for filling the NaN values in the 'language' column here
movies.language.fillna('English')

0       English
1       English
2       English
3       English
4       English
         ...   
3848    English
3849    English
3850    Spanish
3851    English
3852    English
Name: language, Length: 3853, dtype: object

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

In [8]:
movies.loc[pd.isnull(movies['language']), ['language']] = 'English'
(movies.language == 'English').sum()

np.int64(3675)

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [13]:
# Write your code for unit conversion here

movies['budget_mil'] = movies['budget'] / 1000000.0
movies['gross_mil'] = movies['gross'] / 1000000.0

movies.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4  Color     Andrew Stanton                   462.0     132.0   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    475.0                   530.0   Samantha Morton   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   640.0   73058679.0          Action|Adventure|Sci-Fi  ...   

  country content_rating       budget  title_year actor_2_facebook_likes  \
0     USA          PG-13  237000000.0      2009.0                  936.0   
1     USA          PG-13  300000000.0      2007.0                 5000.0   
2      UK          PG-13  245000000.0      2015.0                  393.0   
3     USA          PG-13  250000000.0      2012.0                23000.0   
4     USA          PG-13  263700000.0      2012.0                  632.0   

   imdb_score aspect_ratio movie_facebook_likes  budget_mil   gross_mil  
0         7.9         1.78                33000       237.0  760.505847  
1         7.1         2.35                    0       300.0  309.404152  
2         6.8         2.35                85000       245.0  200.074175  
3         8.5         2.35               164000       250.0  448.130642  
4         6.6         2.35                24000       263.7   73.058679  

[5 rows x 30 columns]

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [15]:
# Write your code for creating the profit column here
movies['net_profit'] = movies['gross_mil'] - movies['budget_mil']

movies.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4  Color     Andrew Stanton                   462.0     132.0   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    475.0                   530.0   Samantha Morton   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   640.0   73058679.0          Action|Adventure|Sci-Fi  ...   

  content_rating       budget  title_year  actor_2_facebook_likes imdb_score  \
0          PG-13  237000000.0      2009.0                   936.0        7.9   
1          PG-13  300000000.0      2007.0                  5000.0        7.1   
2          PG-13  245000000.0      2015.0                   393.0        6.8   
3          PG-13  250000000.0      2012.0                 23000.0        8.5   
4          PG-13  263700000.0      2012.0                   632.0        6.6   

   aspect_ratio movie_facebook_likes budget_mil   gross_mil  net_profit  
0          1.78                33000      237.0  760.505847  523.505847  
1          2.35                    0      300.0  309.404152    9.404152  
2          2.35                85000      245.0  200.074175  -44.925825  
3          2.35               164000      250.0  448.130642  198.130642  
4          2.35                24000      263.7   73.058679 -190.641321  

[5 rows x 31 columns]

In [16]:
# Write your code for sorting the dataframe here
movies.sort_values('net_profit', ascending=False)

color     director_name  num_critic_for_reviews  duration  \
0     Color     James Cameron                   723.0     178.0   
28    Color   Colin Trevorrow                   644.0     124.0   
25    Color     James Cameron                   315.0     194.0   
2704  Color      George Lucas                   282.0     125.0   
2748  Color  Steven Spielberg                   215.0     120.0   
...     ...               ...                     ...       ...   
2147  Color   Katsuhiro Ôtomo                   105.0     103.0   
2136  Color    Hayao Miyazaki                   174.0     134.0   
2693  Color      Lajos Koltai                    73.0     134.0   
3282  Color    Chan-wook Park                   202.0     112.0   
2678  Color      Joon-ho Bong                   363.0     110.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
28                      365.0                  1000.0          Judy Greer   
25                        0.0                   794.0        Kate Winslet   
2704                      0.0                   504.0       Peter Cushing   
2748                  14000.0                   548.0         Dee Wallace   
...                       ...                     ...                 ...   
2147                     78.0                   101.0  Robin Atkin Downes   
2136                   6000.0                   745.0  Jada Pinkett Smith   
2693                     45.0                     0.0     Péter Fancsikai   
3282                      0.0                    38.0        Yeong-ae Lee   
2678                    584.0                    74.0        Kang-ho Song   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
28                    3000.0  652177271.0   
25                   29000.0  658672302.0   
2704                 11000.0  460935665.0   
2748                   861.0  434949459.0   
...                      ...          ...   
2147                   488.0     410388.0   
2136                   893.0    2298191.0   
2693                     9.0     195888.0   
3282                   717.0     211667.0   
2678                   629.0    2201412.0   

                                                 genres  ... content_rating  \
0                       Action|Adventure|Fantasy|Sci-Fi  ...          PG-13   
28                     Action|Adventure|Sci-Fi|Thriller  ...          PG-13   
25                                        Drama|Romance  ...          PG-13   
2704                    Action|Adventure|Fantasy|Sci-Fi  ...             PG   
2748                                      Family|Sci-Fi  ...             PG   
...                                                 ...  ...            ...   
2147  Action|Adventure|Animation|Family|Sci-Fi|Thriller  ...          PG-13   
2136                        Adventure|Animation|Fantasy  ...          PG-13   
2693                                  Drama|Romance|War  ...              R   
3282                                        Crime|Drama  ...              R   
2678                         Comedy|Drama|Horror|Sci-Fi  ...              R   

            budget  title_year  actor_2_facebook_likes imdb_score  \
0     2.370000e+08      2009.0                   936.0        7.9   
28    1.500000e+08      2015.0                  2000.0        7.0   
25    2.000000e+08      1997.0                 14000.0        7.7   
2704  1.100000e+07      1977.0                  1000.0        8.7   
2748  1.050000e+07      1982.0                   725.0        7.9   
...            ...         ...                     ...        ...   
2147  2.127520e+09      2004.0                   336.0        6.9   
2136  2.400000e+09      1997.0                   851.0        8.4   
2693  2.500000e+09      2005.0                     2.0        7.1   
3282  4.200000e+09      2005.0                   126.0        7.7   
2678  1.221550e+10      200

In [17]:
movies.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4  Color     Andrew Stanton                   462.0     132.0   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    475.0                   530.0   Samantha Morton   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   640.0   73058679.0          Action|Adventure|Sci-Fi  ...   

  content_rating       budget  title_year  actor_2_facebook_likes imdb_score  \
0          PG-13  237000000.0      2009.0                   936.0        7.9   
1          PG-13  300000000.0      2007.0                  5000.0        7.1   
2          PG-13  245000000.0      2015.0                   393.0        6.8   
3          PG-13  250000000.0      2012.0                 23000.0        8.5   
4          PG-13  263700000.0      2012.0                   632.0        6.6   

   aspect_ratio movie_facebook_likes budget_mil   gross_mil  net_profit  
0          1.78                33000      237.0  760.505847  523.505847  
1          2.35                    0      300.0  309.404152    9.404152  
2          2.35                85000      245.0  200.074175  -44.925825  
3          2.35               164000      250.0  448.130642  198.130642  
4          2.35                24000      263.7   73.058679 -190.641321  

[5 rows x 31 columns]

In [18]:
top10 = movies[:10]

top10

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4  Color     Andrew Stanton                   462.0     132.0   
5  Color          Sam Raimi                   392.0     156.0   
6  Color       Nathan Greno                   324.0     100.0   
7  Color        Joss Whedon                   635.0     141.0   
8  Color        David Yates                   375.0     153.0   
9  Color        Zack Snyder                   673.0     183.0   

   director_facebook_likes  actor_3_facebook_likes       actor_2_name  \
0                      0.0                   855.0   Joel David Moore   
1                    563.0                  1000.0      Orlando Bloom   
2                      0.0                   161.0       Rory Kinnear   
3                  22000.0                 23000.0     Christian Bale   
4                    475.0                   530.0    Samantha Morton   
5                      0.0                  4000.0       James Franco   
6                     15.0                   284.0       Donna Murphy   
7                      0.0                 19000.0  Robert Downey Jr.   
8                    282.0                 10000.0   Daniel Radcliffe   
9                      0.0                  2000.0       Lauren Cohan   

   actor_1_facebook_likes        gross  \
0                  1000.0  760505847.0   
1                 40000.0  309404152.0   
2                 11000.0  200074175.0   
3                 27000.0  448130642.0   
4                   640.0   73058679.0   
5                 24000.0  336530303.0   
6                   799.0  200807262.0   
7                 26000.0  458991599.0   
8                 25000.0  301956980.0   
9                 15000.0  330249062.0   

                                              genres  ... content_rating  \
0                    Action|Adventure|Fantasy|Sci-Fi  ...          PG-13   
1                           Action|Adventure|Fantasy  ...          PG-13   
2                          Action|Adventure|Thriller  ...          PG-13   
3                                    Action|Thriller  ...          PG-13   
4                            Action|Adventure|Sci-Fi  ...          PG-13   
5                           Action|Adventure|Romance  ...          PG-13   
6  Adventure|Animation|Comedy|Family|Fantasy|Musi...  ...             PG   
7                            Action|Adventure|Sci-Fi  ...          PG-13   
8                   Adventure|Family|Fantasy|Mystery  ...             PG   
9                            Action|Adventure|Sci-Fi  ...          PG-13   

        budget  title_year  actor_2_facebook_likes imdb_score  aspect_ratio  \
0  237000000.0      2009.0                   936.0        7.9          1.78   
1  300000000.0      2007.0                  5000.0        7.1          2.35   
2  245000000.0      2015.0                   393.0        6.8          2.35   
3  250000000.0      2012.0                 23000.0        8.5          2.35   
4  263700000.0      2012.0                   632.0        6.6          2.35   
5  258000000.0      2007.0                 11000.0        6.2          2.35   
6  260000000.0      2010.0                   553.0        7.8          1.85   
7  250000000.0      2015.0                 21000.0        7.5          2.35   
8  250000000.0      2009.0                 11000.0        7.5          2.35   
9  250000000.0      2016.0                  4000.0        6.9          2.35   

  movie_facebook_likes budget_mil   gross_mil  net_profit  
0                33000      237.0  760.505847  523.505847  
1                    0      300.0  309.404152    9.404152  
2                85000      245.0  200.074175  -44.925825  
3               164000      250.0  448.130642  198.130642  
4            

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

In [29]:
top10.movie_title

0                                      Avatar 
1    Pirates of the Caribbean: At World's End 
2                                     Spectre 
3                       The Dark Knight Rises 
4                                 John Carter 
5                                Spider-Man 3 
6                                     Tangled 
7                     Avengers: Age of Ultron 
8      Harry Potter and the Half-Blood Prince 
9          Batman v Superman: Dawn of Justice 
Name: movie_title, dtype: object

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [ ]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'


#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [ ]:
# Write your code for creating three new dataframes here
Meryl_Streep = # Include all movies in which Meryl_Streep is the lead

In [ ]:
Leo_Caprio = # Include all movies in which Leo_Caprio is the lead

In [ ]:
Brad_Pitt = # Include all movies in which Brad_Pitt is the lead

In [ ]:
# Write your code for combining the three dataframes here
Combined = 

In [ ]:
# Write your code for grouping the combined dataframe here


In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>